src/wallet/api/wallet.cpp

```diff
@@ -1324,7 +1326,12 @@ void WalletImpl::setSubaddressLabel(uint32_t accountIndex, uint32_t addressIndex
 
 MultisigState WalletImpl::multisig() const {
     MultisigState state;
-    state.isMultisig = m_wallet->multisig(&state.isReady, &state.threshold, &state.total);
+    tools::wallet2::multisig_status status{m_wallet->get_multisig_status()};
```

Comment:
These structs are duplicates, and the naming schemes are different :/

Is there somewhere `monero::MultisigState` can be placed to so that `wallet2` can re-use it? The problem seems to be that `src/wallet` and `src/wallet/api` aren't supposed to include header files from each other _in the header_ (only in the source of `src/wallet/api`).

Reply:
I am pretty confused about the purpose/use of `src/wallet/api`.

The correct architecture should be like this:
- multisig should be in `namespace cryptonote { namespace multisig { } }`
- wallets should own a `cryptonote::multisig::multisig_account` class object that handles all multisig account operations
- `cryptonote::multisig::multisig_account` should have a member function `get_status()` that returns a `cryptonote::multisig::multisig_account_status` struct. Presumably, `WalletImpl` would also be able to use that struct.

Right now, `wallet2` haphazardly owns a lot of multisig account details. That's why I added `multisig_status` to `wallet2`.

I think I'll move `src/wallet/wallet2 multisig_account -> src/multisig/multisig multisig_account_status` so we are a little closer to the ideal architecture.